File: A_Farmer_John_s_Challenge.py
def solve():
    n, k = map(int, input().split())

    if k > 1 and k != n:
        print(-1)
        return

    ans = [2] * n
    for i in range(k):
        ans[n - i - 1] = 1

    for i in ans:
        print(i, end=" ")
    print()

t = int(input())
for _ in range(t):
    solve()

File: A_Farmer_John_s_Challenge.cpp
#include <iostream>
#include <vector>

using namespace std;

void solve() {
    int n, k;
    cin >> n >> k;

    if (k > (n + 1) / 2) {
        cout << -1 << endl;
        return;
    }

    vector<int> a(n);

    for (int i = 0; i < n; ++i)
        a[i] = i + 1;

    for (int i = 1; i <= k; ++i)
        a[n - i] = i;

    for (int i = 0; i < n; ++i)
        cout << a[i] << " ";
    cout << endl;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        solve();
    }

    return 0;
}
